### PR TITLE
Add E2 extension menu

### PR DESCRIPTION
**Pull request by AbigailBuccaneer**
Would've had a fancy first-run popup but Garry and his GUI system hates me.

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/http.lua [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	     (McLovin)
 */
 
-E2Lib.RegisterExtension( "http", false )
+E2Lib.RegisterExtension( "http", false, "Lets E2 chips make web requests to the real internet.", "Allows any E2 to make your server make arbitrary HTTP GET requests to any site. It can use this to make HTTP requests to any IP address inside your local network." )
```
Comment (Divran) [inline lua/entities/gmod_wire_expression2/core/http.lua]: This warning won't be entirely true ater we merge the client side http stuff

**Comment by Divran** [comment]
I demand pictures

**Comment by oldmud0** [comment]
> I demand pictures

File: lua/entities/gmod_wire_expression2/core/http.lua (hunk 1 above)
Comment (AbigailBuccaneer) [inline lua/entities/gmod_wire_expression2/core/http.lua]: Well, whichever one gets merged last can clean it up.

**Comment by AbigailBuccaneer** [comment]
It originally was a separate DFrame which could be opened with a console command, but that was a bit pointless.

**Comment by Divran** [comment]
It could also be placed in the wire tab, since we already have a few similar options menus there.

**Comment by oldmud0** [comment]
Some extensions still have unexplained security issues. Example: `holo` can be used to blind players, `file` has no size limit, `console` can run any command at all, `propcore` can be used for prop pushing (killing players), ...

**Comment by Divran** [comment]
console can't run any console command. Some are blocked by garry (and there's a whitelist that the client can change). Propcore can't be used for prop pushing. You're thinking of applyForce (which isn't an extension and can't be disabled, maybe it should be?)

**Comment by oldmud0** [comment]
That still leaves holo and file though..

**Comment by Divran** [comment]
you can block other people's holograms from rendering for you, which prevents blinding. Not sure if anyone knows about those console commands though.
Yep file doesn't have size limits, but you can only create files on your own computer, which is only a problem if you use someone else's E2 and they deliberately create a huge file on your computer with it.

**Comment by bigdogmat** [comment]
file has a default limit of 300kb right?

**Comment by AbigailBuccaneer** [comment]
Would anybody object if I pulled this and then people can make PRs for extensions explaining security issues?

**Comment by Divran** [comment]
>file has a default limit of 300kb right?
EDIT: By default it has a 300kb limit.
https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/files.lua#L7

yeah but you can just make more files

**Comment by AbigailBuccaneer** [comment]
If extensions need more description / warning please make new issues / PRs.

**Comment by oldmud0** [comment]
BOOM

**Comment by TomyLobo** [comment]
I picture demands!

> Would anybody object if I pulled this and then people can make PRs for extensions explaining security issues?

Good idea, except if it's a first-time popup thingy. Didn't look at it in detail.

**Comment by AbigailBuccaneer** [comment]
There's no first-time popup.

**Pull request merged**